File: ruby/sparql1.rb
# Ruby querying of the BioPortal triplestore using standard Ruby libraries and the JSON rubygem

require 'rubygems'
require 'json'
require 'open-uri'
require 'cgi'

apikey = ""
endpoint = "http://sparql.bioontology.org/sparql/"
query = <<-EOS
PREFIX omv: <http://omv.ontoware.org/2005/05/ontology#>

SELECT ?ont ?name ?acr
WHERE { ?ont a omv:Ontology;
             omv:acronym ?acr;
             omv:name ?name .
} LIMIT 100
EOS

# Returns a JSON object with the results of a SPARQL query
def query(query, apikey, endpoint, accept = "application/json")
	json = open("#{endpoint}?query=#{CGI.escape(query)}&apikey=#{apikey}", "Accept" => accept).read
	JSON.parse(json)

end

puts JSON.pretty_generate(query(query, apikey, endpoint))

File: python/sparql1.py
""" Simple Python script to query "http://sparql.bioontology.org/sparql/"
    No extra libraries required.
"""

import json
import urllib2
import urllib
import traceback
import sys 

def query(q,apikey,epr,f='application/json'):
    """Function that uses urllib/urllib2 to issue a SPARQL query.
       By default it requests json as data format for the SPARQL resultset"""

    try:
        params = {'query': q, 'apikey': apikey}
        params = urllib.urlencode(params)
        opener = urllib2.build_opener(urllib2.HTTPHandler)
        request = urllib2.Request(epr+'?'+params)
        request.add_header('Accept', f)
        request.get_method = lambda: 'GET'
        url = opener.open(request)
        return url.read()
    except Exception, e:
        traceback.print_exc(file=sys.stdout)
        raise e

if __name__ == "__main__":
    sparql_service = "http://sparql.bioontology.org/sparql/"

    #To get your API key register at http://bioportal.bioontology.org/accounts/new
    api_key = "YOUR API KEY HERE"

    #Some sample query.
    query_string = """ 
PREFIX omv: <http://omv.ontoware.org/2005/05/ontology#>

SELECT ?ont ?name ?acr
WHERE { ?ont a omv:Ontology;
             omv:acronym ?acr;
             omv:name ?name .
} 
"""
    json_string = query(query_string, api_key, sparql_service)
    resultset=json.loads(json_string)

    #Printing the json object.
    print json.dumps(resultset,indent=1)
